File: jelly-runtime/src/store/dapp/access/nft_owner.ts
import { Chain } from './chain';

export interface NFTOwner {
    chain: Chain;
    address: string; // Specify NFT contract
    token_id?: string; // Any NFT or specified NFT
}

export interface VerifiedNFTOwner {
    chain: Chain;
    address: string; // Specify NFT contract
    token_id?: string; // Any NFT or specified NFT

    // =============== Verification information ===============
    identity: string; // Identity address
    message: string;
    signature: string;
}

File: jelly-model/src/store/dapp/access/nft_owner.rs
use serde::{Deserialize, Serialize};

/// NFT owner
#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
pub struct NFTOwner {
    /// chain
    pub chain: super::chain::Chain,
    /// Specify NFT contract
    pub address: String,
    /// Any NFT or specified NFT
    #[serde(skip_serializing_if = "Option::is_none")]
    pub token_id: Option<String>,
}

/// NFT owner
#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
pub struct VerifiedNFTOwner {
    /// chain
    pub chain: super::chain::Chain,
    /// Specify NFT contract
    pub address: String,
    /// Any NFT or specified NFT
    #[serde(skip_serializing_if = "Option::is_none")]
    pub token_id: Option<String>,

    // =============== Verification information ===============
    /// Identity address
    pub identity: String,
    /// Signature information
    pub message: String,
    /// signature
    pub signature: String,
}
